app/controllers/discourse_ai/admin/ai_tools_controller.rb

```diff
@@ -17,18 +17,20 @@ def show
       end
 
       def create
-        ai_tool = AiTool.new(ai_tool_params)
+        ai_tool = AiTool.new(ai_tool_params.except(:rag_uploads))
         ai_tool.created_by_id = current_user.id
 
         if ai_tool.save
+          RagDocumentFragment.link_target_and_uploads(ai_tool, attached_upload_ids)
           render_serialized(ai_tool, AiCustomToolSerializer, status: :created)
         else
           render_json_error ai_tool
         end
       end
 
       def update
-        if @ai_tool.update(ai_tool_params)
+        if @ai_tool.update(ai_tool_params.except(:rag_uploads))
```

Comment:
Im surprise a lot of this code is using non ! methods. same for save above.

What happens if `@ai.tool.update` fails?

Reply:
yeah it works, it does a render json error that consumes the error.